src/_guides/openapi/design-first-rails.md

```diff
@@ -0,0 +1,200 @@
+---
+
+Ruby on Rails developers are blessed with a bunch of great [OpenAPI](https://spec.openapis.org/oas/latest.html) tooling, and can use either of the API Code-First workflow which was popular for a long time, or the follow the newer API Design-first workflow. 
+
+Instead of writing loads of code and sprinkling in some metadata later to create docs, the design-first workflow assumes you create the OpenAPI descriptions before writing any code at all. Once you have the OpenAPI description documents saved in your repository, you can leverage it at every step of the API lifecycle, to produce mock APIs for clients to test assumptions with, build client libraries without writing any code, make really effective contract testing, and even generate backend code to get the application teams started once the contract is all signed off.
+
+This guide is going to look at two specific parts of the API design-first workflow that are most helpful to documentation, and show how to set it up in Rails: request validation automatically, and contract testing responses.
+)
+- [Getting OpenAPI \& Bump.sh Setup](#getting-openapi--bumpsh-setup)
+- [Request Validation powered by OpenAPI](#request-validation-powered-by-openapi)
+- [Contract Testing with OpenAPI](#contract-testing-with-openapi)
+- [Sample Code](#sample-code)
+
+## Getting OpenAPI & Bump.sh Setup
+
+The API design-first workflow means you'll need to create your OpenAPI description before you start writing all your code, so if you don't have an `openapi.yaml` already that is probably the first step. You can use a wide variety of [graphical editors](https://openapi.tools/#gui-editors), [text editors](https://openapi.tools/#text-editors), or [traffic sniffing](/guides/openapi/code-first/#traffic-sniffing) to generate this OpenAPI, and there is lots of documentation and guides to help you. 
+
+Alternatively you can grab some sample OpenAPI from the API Guru Marketplace, and click JSON or YAML to download their OpenAPI descriptions.
+
+Either way, one you have some an OpenAPI description document, pop it into your Git repository somewhere like `api/openapi.yaml`. 
+
+Building an API for a bunch of clients is always a tricky one, but by deploying the documentation first you can see if people like the look of the API before you waste a bunch of time building it. Then as you progress through, especially if you are adding these tools to an existing codebase, you will continue to find mistakes in your OpenAPI or your actual API code, improving both as you go until you have a perfect match that will never again be broken, solving the "docs vs code" drift problem, and every fix will be deployed to Bump with each commit/merge.
+
+```bash
+$ bump deploy api/openapi.yaml \
+  --doc my-documentation-name \
+  --token my-documentation-token
+
+* Your new documentation version will soon be ready at https://bump.sh/philsturgeon/doc/rails-design-first
```

Comment:
Do you want to update this one to the new repo under the Bump.sh GitHub Org?

Reply:
There's a new Github repo: https://github.com/bump-sh/rails-hello-openapi but that's not yet deploying to any bump docs, and I don't have enough settings to make it so.

I've made my own Bump Examples organization and will make new projects over there, then deploy each of the three sample projects I've made so far.

Reply:
Updated links here and elsewhere to point to https://bump.sh/bump-examples/hub/code-samples/